wallet/db.c

```diff
@@ -393,7 +392,7 @@ static struct migration dbmigrations[] = {
     /* Now make sure we have the lower bound block with the first_blocknum
      * height. This may introduce a block with NULL height if we didn't have any
      * blocks, remove that in the next. */
-    {SQL("INSERT OR IGNORE INTO blocks (height) VALUES ((SELECT "
+    {SQL("INSERT INTO blocks (height) VALUES ((SELECT "
```

Comment:
Both should be happy with INSERT .... ON CONFLICT IGNORE

Reply:
Nope, postgres [doesn't like `IGNORE`, it calls it `DO NOTHING`](https://www.postgresql.org/docs/9.5/sql-insert.html). How I love these little nuggets :wink: I'll add a rewrite rule for it

Reply:
Nope, turns out `sqlite3` doesn't support `ON CONFLICT` statements up until version 3.24 so this is not portable in any case, unless we want to make for a really weird rewrite-rule that transforms a `INSERT OR IGNORE` into a `INSERT INTO ...loads of other stuff... ON CONFLICT DO NOTHING`.

In this case I don't think it's worth it, since it predates v0.6 :wink:

Reply:
We should make this explicit then, I think (separate patch at the end).  Remove early migrations, complain if db is before that, and replace them with nicer db statements which are only used on initializing the db.

Reply:
Yech, that's admitting defeat! I'll give it another go.